File: pinyougou-manager-web/src/main/webapp/js/service/BrandService.js
//品牌服务
app.service("brandService",function ($http) {
    this.find=function () {
        return $http.get("../brand/findAll.do");
    };
    this.findPage=function (page,size) {
        return $http.get('../brand/findPage.do?page='+page+'&size='+size);
    };
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    };
    this.add=function (iphone) {
        return $http.post('../brand/add.do',iphone)
    };
    this.update=function (iphone) {
        return $http.post('../brand/update.do',iphone)
    };
    this.delete=function (ids) {
        return $http.get("../brand/delete.do?id="+ids);
    };
    this.search=function (page,size,searchIphone) {
        return  $http.post('../brand/search.do?page='+page+'&size='+size,searchIphone);
    }
    this.selectOption=function () {
        return $http.get('../brand/selectOption.do')
    }
});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageRresult;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 */
public interface BrandService {
    /**
     * 查询全部数据
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 返回分页列表
     * @return
     */
    public PageRresult findPage(int pageNum,int pageSize);
    /**
     * 增加数据
     */
    public void add(TbBrand tbBrand);

    /**
     * 单个查询
     */
    public TbBrand findOne(Long id);

    /**
     * 修改方法
     */
    public void update(TbBrand tbBrand);
    /**
     * 删除
     */
    public void delete(Long [] ids);
    /**
     * 查找返回分页
     */
    public PageRresult findPage(TbBrand tbBrand,int pageNum,int pageSize);

    public List<Map> selectOption();
}
